templates/default/origin-dns.conf.erb

```diff
@@ -1,5 +1,7 @@
-strict-order
 no-resolv
 domain-needed
-server=/cluster.local/<%=node['cookbook-openshift3']['openshift_common_first_svc_ip'] %>
-
+no-negcache
+max-cache-ttl=1
+enable-dbus
+bind-interfaces
+listen-address=<%= node['cookbook-openshift3']['openshift_common_ip'] %>
```

Comment:
This last line conflicts on my VMs because dnsmasq is forced to listen to `node['cookbook-openshift3']['openshift_common_ip']` (192.168.122.220 on my VM) while /etc/resolv.conf tells the system to use the IP of the default route as server (which on my system is 10.0.2.15). If unable to have the same IP in /etc/resolv.conf and in /etc/dnsmasq.d/origin-dns.conf then better have /etc/dnsmasq.d/origin-dns.conf listen to all interfaces.

See below:
```sh
[root@master ~]# ping www.google.fr
ping: www.google.fr: Name or service not known  # <---- broken
[root@master ~]# cat /etc/resolv.conf 
# nameserver updated by /etc/NetworkManager/dispatcher.d/99-origin-dns.sh
# Generated by NetworkManager
options single-request-reopen
nameserver 10.0.2.15    # <---- this is the default route IP, using the NAT interface of my VM
[root@master ~]# netstat -ntlp | grep :53
tcp        0      0 192.168.33.220:53       0.0.0.0:*               LISTEN      15555/dnsmasq       # <---- dnsmasq is no longer listening on all interfaces, only on the openshift_common_ip (which is != than the server in /etc/resolv.conf)
[root@master ~]# cat /etc/dnsmasq.d/origin-dns.conf 
no-resolv
domain-needed
no-negcache
max-cache-ttl=1
enable-dbus
bind-interfaces
listen-address=192.168.33.220 # <------ this is the problem, just remove the line and it'll be fine.
```

Reply:
hmmm I can see that issue. We'll need to put more logic...

Reply:
Will need to correct that value as you won't be able to bind on any port 0.0.0.0 (Will be clashing with dnsmasq). They have change in 3.6 the dns style https://github.com/openshift/origin/blob/master/pkg/cmd/server/api/types.go#L187-L212